File: week2/atakantekatan/3. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        word = ""
        ans = 0
        for letter in s:
            if letter not in word:
                word += letter
            else:
                ans = max(ans,len(word))
                while letter in word:
                    word = word[1:]
                word += letter
        return max(ans,len(word))

File: week2/Eren-Taskiran/3-longest-substring-without-repeating-characters.java
class Solution {
public int lengthOfLongestSubstring(String s) {
    int left = 0, right = 0, maxLength = 0;
    Set<Character> characters = new HashSet<>();

    while (right < s.length()) {
        if (!characters.contains(s.charAt(right))) {
            characters.add(s.charAt(right));
            maxLength = Math.max(maxLength, right - left + 1);
            right++;
        } else {
            characters.remove(s.charAt(left));
            left++;
        }
    }

    return maxLength;
}}